plugins/PHP/Plugins/PerRequestPlugins.php

```diff
@@ -66,16 +66,20 @@ private function __construct()
 
         if(isset($_SERVER['REMOTE_ADDR'])) {
             pinpoint_add_clue(PP_REQ_CLIENT, $_SERVER["REMOTE_ADDR"]);
+        } elseif (($hostname = gethostname()) !== false) {
+            pinpoint_add_clue(PP_REQ_CLIENT, $hostname);
         }
 
         if(isset($_SERVER['HTTP_HOST'])) {
             pinpoint_add_clue(PP_REQ_SERVER, $_SERVER["HTTP_HOST"]);
+        } elseif(($pid = getmypid()) !== false) {
+            pinpoint_add_clue(PP_REQ_SERVER, sprintf("[pid:%d]", $pid));
```

Comment:
PP_REQ_SERVER : current server information

Reply:
PerRequestPlugins, we just want to show some commonly used fields. 

While,  you can extend your own `PerRequestPlugins`. 

``` php 

define('PP_REQ_PLUGINS', '\Plugins\yourOwnPlugins');

class yourOwnPlugins extends PerRequestPlugins

```

pinpoint_add_clue,  The new key-value pair will replace the old key-value pair.